### PR TITLE
[NetManager] Components Lazy loading 

### PR DESCRIPTION
**Pull request by ja-odur**
#### Summary of Changes (What does this PR do?)
- Implement component lazy loading to improve to page load time

#### Is this change ready to hit production in its current state?
- [ ] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [ ] I consider this code done
- [ ] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Fetch the branch locally
* Install dependencies `npm install`
* Start the application `npm run stage-mac` (macOS) or `npm run stage-pc` (Windows platform)
##### Testing on Chrome Browser
* Load the application on Chrome
* Open `Chrome developer console`. 
* Navigate to the network panel and check the disable cache option. To view the changes, may need to to use `slow3G` option.
* Reload the app with the console still open.
* Navigate any any other page using the sidebar tab, you should see the fallback loader while the component resources are being fetched from the server.

#### What are the relevant tickets?
- [PLAT-157](https://airqoteam.atlassian.net/browse/PLAT-157)

#### Screenshots (optional)

### CODE REVIEW

**Review by danielogen — APPROVED** [review]
Look good...great reduction in the bundle size

**Review by lilly-muyama — APPROVED** [review]

**Review by sserurich — COMMENTED** [review]
It works perfectly well, I could see(observe) the changes mentioned in the PR. however every time I navigated to different section(page) console errors were being thrown.. not sure whether it's on my side, but @ja-odur , can confirm and incase it's fine, respond to this and I will approve.
image

**Comment by ja-odur** [comment]
> It works perfectly well, I could see(observe) the changes mentioned in the PR. however every time I navigated to different section(page) console errors were being thrown.. not sure whether it's on my side, but @ja-odur , can confirm and incase it's fine, respond to this and I will approve.
> image

@sserurich The app generally has some console warnings but let me try to recreate the issue to be sure that it's not fatal.

File: netmanager/src/App.js [hunk 1]
```diff
@@ -102,65 +100,56 @@ const App = () => {
               />
```
Comment (Baalmart) [inline netmanager/src/App.js:100]: Side note: we shall need to transfer this route logic to the routers directory accordingly. Thanks.

File: netmanager/src/views/stateConnectors/containerConnector.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+// map state from store to props
```
Comment (Baalmart) [inline netmanager/src/views/stateConnectors/containerConnector.js:1]: This is a good refactor, the usage of descriptive names for all of our files/directories is extremely welcome.

**Review by Baalmart — COMMENTED** [review]
The refactoring done is awesome, I have also seen the reduction in bundle size.

But not yet so confident about the loading speed whenever I navigate the side bar items, could I be missing something?

File: netmanager/src/views/stateConnectors/containerConnector.js (hunk 2 above)
Comment (ja-odur) [reply netmanager/src/views/stateConnectors/containerConnector.js:1]: Coming up with good names takes sometime. I prefer to do it as last thing 😄

File: netmanager/src/App.js (hunk 1 above)
Comment (ja-odur) [reply netmanager/src/App.js:100]: This component holds these high level components (components that require the layout component). The routes are quite many but removing them will leave this component empty or unnecessary. Not sure which approach is best here but leaving as it is may be the best.

**Comment by ja-odur** [comment]
> The refactoring done is awesome, I have also seen the reduction in bundle size.
> 
> But not yet so confident about the loading speed whenever I navigate the side bar items, could I be missing something?

@Baalmart if you were following the instructions in the testing section, we intentionally set the network to slow so that we can observe the fetching and loading of components when you navigate through the app using the sidebar tabs. The goal is not to return the current overall bundle size but to cut the initial load size (currently from 3.8mb to 1.7mb) in-order to speed up the initial load of the application (by only loading the core components of the app). The other components are loaded on demand.

File: netmanager/src/views/stateConnectors/containerConnector.js (hunk 2 above)
Comment (ja-odur) [reply netmanager/src/views/stateConnectors/containerConnector.js]: @Baalmart This is fixed

File: netmanager/src/App.js (hunk 1 above)
Comment (Baalmart) [reply netmanager/src/App.js:100]: Cool, thanks.

**Comment by Baalmart** [comment]
> > The refactoring done is awesome, I have also seen the reduction in bundle size.
> > But not yet so confident about the loading speed whenever I navigate the side bar items, could I be missing something?
> 
> @Baalmart if you were following the instructions in the testing section, we intentionally set the network to slow so that we can observe the fetching and loading of components when you navigate through the app using the sidebar tabs. The goal is not to return the current overall bundle size but to cut the initial load size (currently from 3.8mb to 1.7mb) in-order to speed up the initial load of the application (by only loading the core components of the app). The other components are loaded on demand.

Okay, sounds reasonable now. Thanks for the clarification, missed the "intentionally set the network to slow". :) Thanks @ja-odur

**Review by Baalmart — APPROVED** [review]
LGTM

**Pull request merged**